packages/number-field/test/number-field.test.js

```diff
@@ -0,0 +1,884 @@
+      numberField.min = -0.02;
+      numberField.max = 0.02;
+      numberField.value = -0.03;
+      numberField.step = 0.01;
+
+      const correctSteps = [-0.02, -0.01, 0, 0.01, 0.02];
+      for (var i = 0; i < correctSteps.length; i++) {
+        increaseButton.click();
+        expect(numberField.value).to.be.equal(String(correctSteps[i]));
+      }
+    });
+
+    it('should correctly calculate the precision with decimal value', () => {
+      numberField.value = 5.1;
+      numberField.step = 0.01;
+
+      increaseButton.click();
+      expect(numberField.value).to.be.equal('5.11');
+    });
+
+    describe('problematic values', () => {
+      it('should correctly increase value', () => {
+        const configs = [
+          { props: { step: 0.001, value: 1.001 }, expectedValue: '1.002' },
+          { props: { step: 0.001, value: 1.003 }, expectedValue: '1.004' },
+          { props: { step: 0.001, value: 1.005 }, expectedValue: '1.006' },
+          { props: { step: 0.001, value: 2.002 }, expectedValue: '2.003' },
+          { props: { step: 0.001, value: 4.004 }, expectedValue: '4.005' },
+          { props: { step: 0.001, value: 8.008 }, expectedValue: '8.009' },
+          { props: { step: 0.01, value: 16.08 }, expectedValue: '16.09' },
+          { props: { step: 0.01, value: 73.1 }, expectedValue: '73.11' },
+          { props: { step: 0.001, value: 1.0131, min: 0.0001 }, expectedValue: '1.0141' }
+        ];
+        const reset = { step: 1, min: undefined, max: undefined, value: '' };
+
+        for (let i = 0; i < configs.length; i++) {
+          const { props, expectedValue } = configs[i];
+          Object.assign(numberField, reset, props);
+          increaseButton.click();
+          expect(numberField.value).to.be.equal(expectedValue);
+        }
+      });
+
+      it('should correctly decrease value', () => {
+        const configs = [
+          { props: { step: 0.01, value: 72.9 }, expectedValue: '72.89' },
+          { props: { step: 0.001, min: 0.0001, value: 1.0031 }, expectedValue: '1.0021' },
+          { props: { step: 0.001, min: 0.0001, value: 1.0051 }, expectedValue: '1.0041' },
+          { props: { step: 0.001, min: 0.0001, value: 1.0071 }, expectedValue: '1.0061' },
+          { props: { step: 0.001, min: 0.0001, value: 1.0091 }, expectedValue: '1.0081' }
+        ];
+        const reset = { step: 1, min: undefined, max: undefined, value: '' };
+
+        for (let i = 0; i < configs.length; i++) {
+          const { props, expectedValue } = configs[i];
+          Object.assign(numberField, reset, props);
+          decreaseButton.click();
+          expect(numberField.value).to.be.equal(expectedValue);
+        }
+      });
+    });
+  });
+
+  describe('no initial value', () => {
+    describe('min is defined and max is undefined', () => {
+      it('should set value to the first step value above zero when min is below zero and increaseButton is clicked', () => {
+        numberField.min = -19;
```

Comment:
~~I wonder if the approach of defining test's circumstances (that is used [above](https://github.com/vaadin/web-components/pull/2279/files#diff-6a4e76f7ebd92a2907188f897c1e37724fc096729eac637534efd537d08b1b48R366-R374)) might be appropriate here as well.~~

**Update:** This part is just a copy-paste from the old version of the NumberField.